pkg/cmd/admin/router/router.go

```diff
@@ -325,6 +325,11 @@ func generateSecretsConfig(cfg *RouterConfig, kClient *kclient.Client,
 		mounts = append(mounts, mount)
 	}
 
+	// When the user provides a DEFAULT_CERT environment variable it must be in pem format.
+	// If the default cert is not provided by the user, we generate one using a service
+	// annotation (see RunCmdRouter()), but that will have a cert and a key that we need to
+	// combine into the pem that haproxy expects
+	var certName = fmt.Sprintf("%s-certs", cfg.Name)
 	if len(defaultCert) > 0 {
 		keys, err := cmdutil.PrivateKeysFromPEM(defaultCert)
```

Comment:
Well, actually extract the private key just to verify that we have a pem (which haproxy is expecting). We don't actually need (or use) the key.
The code talks about a default cert when it really means a default pem. There are a lot of comments here that explain the confusing variable naming.